src/app/screens/Enable/LiquidEnable.tsx

```diff
@@ -0,0 +1,43 @@
+import { useEffect, useState } from "react";
+import LiquidEnableComponent from "~/app/components/Enable/LiquidEnable";
+import Onboard from "~/app/components/onboard";
+import { useAccount } from "~/app/context/AccountContext";
+import api from "~/common/lib/api";
+import type { OriginData } from "~/types";
+
+type Props = {
+  origin: OriginData;
+};
+
+export default function LiquidEnable(props: Props) {
+  const { account } = useAccount();
+  const [hasMnemonic, setHasMnemonic] = useState(false);
+
+  useEffect(() => {
+    async function fetchAccountAndSetComponent() {
```

Comment:
can this method be renamed? that's not what it does

What about `fetchAccountInfo` or something? (issue is for all providers)